scripts/zones/Horlais_Peak/mobs/Evil_Oscar.lua

```diff
@@ -2,9 +2,83 @@
 -- Area: Horlais Peak
 --  Mob: Evil Oscar
 -- KSNM30
+--
+-- DONE: If you are out of range of EBB, the move will be lost
+-- TODO: There should be a "No targets within range" message
+--
+-- ['Evil Oscar-17346750'] = {
+--     [4] = {
+--         -- NOTE: This looks to be the no target message
+--         [0]    = {['name']="",                     ['category']=4,  ['id']=0,    ['animation']=508,  ['message']=76,  },
+--     },
+--     [11] = {
+--         [317]  = {['name']="Vampiric Lash",        ['category']=11, ['id']=317,  ['animation']=61,   ['message']=31,  },
+--         [320]  = {['name']="Sweet Breath",         ['category']=11, ['id']=320,  ['animation']=64,   ['message']=185, },
+--         [1332] = {['name']="Extremely Bad Breath", ['category']=11, ['id']=1332, ['animation']=63,   ['message']=406, },
+--     },
+-- },
+-----------------------------------
+local ID = require("scripts/zones/Horlais_Peak/IDs")
 -----------------------------------
 local entity = {}
 
+local EXTREMELY_BAD_BREATH = 1332
+
+local sendMessage = function(players)
+    for _, member in pairs(players) do
+        member:messageSpecial(ID.text.EVIL_OSCAR_BEGINS_FILLING)
+    end
+end
+
+local evilOscarFillsHisLungs
```

Comment:
While not incorrect, is there a reason for declaring the variable before assignment on the next line?  Other than that, looks good to me. 👍

Reply:
Good question:

This is to make sure the variable `evilOscarFillsHisLungs` exists fully in the scope of this file BEFORE the function is declared. That way the function is able to call `evilOscarFillsHisLungs` recursively. If it wasn't declared this way, the function wouldn't be able to "see" itself for repeated calls.

Reply:
Further examples (for future reference):

This won't work, because test isn't defined for the function body:
```lua
local test = function(num)
    print(num)
    if num >= 10 then
        return
    else
        test(num + 1)
    end
end
test(0)
```

This works because test is defined (even though its nil to start with):
```lua
local test
test = function(num)
    print(num)
    if num >= 10 then
        return
    else
        test(num + 1)
    end
end
test(0)
```